packages/web-components/src/components/time-picker/time-picker.ts

```diff
@@ -0,0 +1,375 @@
+  get value() {
+    return this._input ? this._input.value : this._value;
+  }
+
+  set value(value) {
+    const oldValue = this._value;
+    this._value = value;
+    this.requestUpdate('value', oldValue);
+    if (this._input) {
+      this._input.value = value;
+    }
+  }
+
+  render() {
+    const {
+      className,
+      disabled,
+      hideLabel,
+      invalid,
+      invalidText,
+      warning,
+      warningText,
+      labelText,
+      placeholder,
+      readonly,
+      maxLength,
+      pattern,
+      size,
+      type,
+      value,
+      _handleInput: handleInput,
+      _handleClick: handleClick,
+      _handleBlur: handleBlur,
+      _handleSlotChange: handleSlotChange,
+    } = this;
+
+    const labelClasses = classMap({
+      [`${prefix}--label`]: true,
+      [`${prefix}--visually-hidden`]: hideLabel,
+      [`${prefix}--label--disabled`]: disabled,
+    });
+
+    const timePickerClasses = classMap({
+      [`${prefix}--time-picker`]: true,
+      [`${prefix}--time-picker--invalid`]: invalid,
+      [`${prefix}--time-picker--warning`]: warning,
+      [`${prefix}--time-picker--readonly`]: readonly,
+      [`${prefix}--time-picker--${size}`]: size,
+      ...(className && { [className]: true }),
+    });
+
+    const inputClasses = classMap({
+      [`${prefix}--time-picker__input-field`]: true,
+      [`${prefix}--text-input`]: true,
+      [`${prefix}--time-picker__input-field-error`]: invalid || warning,
+      ...(className && { [className]: true }),
+    });
+    const label = labelText
+      ? html`<label class="${labelClasses}">${labelText}</label>`
+      : null;
+
+    return html`
+      <div class="${prefix}--form-item ">
+        ${label}
+        <div class="${timePickerClasses}">
+          <div class="${prefix}--time-picker__input">
+            <input
+              class="${inputClasses}"
+              ?data-invalid="${invalid}"
+              ?disabled="${disabled}"
+              maxlength="${ifNonEmpty(maxLength)}"
+              name="${ifNonEmpty(this.name)}"
+              pattern="${ifNonEmpty(pattern)}"
+              placeholder="${ifNonEmpty(placeholder)}"
+              ?readonly="${readonly}"
+              type="${ifNonEmpty(type)}"
+              .value="${value}"
+              @input="${handleInput}"
+              @click="${handleClick}"
+              @blur="${handleBlur}" />
+            ${invalid || warning
+              ? html`
+                  <div class="${prefix}--time-picker__error__icon">
+                    ${invalid
+                      ? WarningFilled16({
+                          class: `${prefix}--checkbox__invalid-icon`,
+                        })
+                      : WarningAltFilled16({
+                          class: `${prefix}--text-input__invalid-icon--warning`,
+                        })}
+                  </div>
+                `
+              : null}
+          </div>
+          <slot
+            name="time-picker-select"
```

Comment:
we might not need a name on this `slot` since it's the only `slot`. Then we could also not add the `slot` info on the `cds-time-picker-select` component